falguard.py

```diff
@@ -61,10 +63,17 @@ def to_dict(self, obj_type=dict):
 
 
 class Validator:
-
     def __init__(self, spec_path):
+        deserializers = {
+            '.json': json.loads,
+            '.yaml': yaml.load,
+            '.yml': yaml.load
+        }
         with open(spec_path, 'r') as f:
-            spec_dict = json.loads(f.read())
+            # pylint: disable=unused-variable
+            filename, ext = os.path.splitext(spec_path)
+            spec_dict = deserializers[ext](f.read())
+        del deserializers
```

Comment:
Nah, let's drop this overoptimisation. Please move deserializers to module scope (as DESERIALIZERS constant) and don't clear it from memory. It will be much easier to understand for newcomers.